### DISCUSSION TITLE
Diagrams in Safari not working probably - Can't find variables

### DISCUSSION THREAD

**Discussion by schleiferjunge** [firefly-iii/firefly-iii#7476] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

By clicking on "Dashboard", I'm getting following errors. They also appear on demo site.
This leads to not showing diagrams or not be able to chose a time range.

![image](https://user-images.githubusercontent.com/52082704/236827694-d4bae69a-f81b-4c21-ab07-0ae4814bccee.png)

Can't find variables: sessionStart
Can't find variables: currencySymbol
Can't find variables: todayText
![image](https://user-images.githubusercontent.com/52082704/236827080-be99b4f5-51bc-49ba-b2a8-f0da26f02bd7.png)

### Debug information

Debug information generated at 2023-05-08 00:00:00 Europe/Berlin for Firefly III version **6.0.9**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.9 |
| Firefly III API | 2.0.1 |
| Build | 466, 29-04-2023 06:55:07 CEST |
| DB version | 19 (exp. 19) |
| PHP | `8.2.4` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Berlin |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 67108864 (64 MB) |
| Interface | apache2handler |
| Default language | de_DE |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-05-01 00:00:00 |
| Session end | 2023-05-31 23:59:59 |
| Session first | 2020-11-26 00:00:00 |
| User ID | 3 |
| User language | de_DE |
| User locale | de_DE |
| Attempt at "de_DE.utf8" | 'de_DE.utf8' |
| Attempt at "de_DE.UTF-8" | 'de_DE.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.4 Safari/605.1.15 |

### Expected behaviour

There should be diagrams.

### Steps to reproduce

1. Login
2. click on dashboard

### Additional info

Could be Safari problem. Chrome works fine.

**Comment by JC5** [comment]
That's weird, thanks for reporting! We are running literally the same browser (Safari + OS) but I'm not seeing this on the demo site.

Are you running any anti-ad extensions, or something else perhaps?

**Reply by schleiferjunge** [reply]
I‘m using pihole as dns forwarder to block some ads network wide. But this happens just in the Safari version for mac. 
The mobil version of Safari works like charme without changing anything in the pihole settings.

Also i checked it with another macbook and got it reproduced. 

After reloading the page one ore two times, the errors are gone and the diagrams appear again.

Let me know if i can provide you more information.

**Reply by JC5** [reply]
Mmm, I'm not seeing this atm. It jumps between working and not working, you say?

**Reply by schleiferjunge** [reply]
Tried to capture it. When Clicking on dashboard the variables are gone. After reloading everything works fine.
https://github.com/firefly-iii/firefly-iii/assets/52082704/6a79dea1-6ead-40f5-b53d-fc6e69651129

I will try it with another Mac and give feedback.

**Reply by JC5** [reply]
I see what you mean, I'm looking for the root cause. You had the same issue locally as well, right?

**Reply by schleiferjunge** [reply]
That's right.
